### ISSUE TITLE
SpawnInsideModifier not resetting on scene reload (with fix)

### ISSUE THREAD

**Issue by demi180** [mapbox/mapbox-unity-sdk#648]
This has been the case since we started working with Mapbox, just now got around to fixing it. For our use we often need to just reload the current scene (easier than trying to reset specific objects).

With previous versions the buildings were also having trouble reloading, but now that seems to be working. But trees and other objects spawned with the SpawnInsideModifier are not properly respawning and instead an exception happens. Because the modifiers are ScriptableObjects, and everything still holds the temp data in SOs instead of in a runtime object.

To recreate, simply take a scene with some tree spawns, add in a script to reload current scene, ex:
`SceneManager.LoadScene ( SceneManager.GetActiveScene ().name );`
My solution was to just safely destroy and clear any objects in this modifier object, like this:
```C#
public override void Initialize()
{
	if (_objects == null || _pool == null)
	{
		_objects = new Dictionary<GameObject, List<GameObject>>();
		_pool = new Queue<GameObject>();
	} else
	{
		foreach ( var pair in _objects )
		{
			foreach ( GameObject go in pair.Value )
				if ( go != null )
					Destroy ( go );
			if ( pair.Key != null )
				Destroy ( pair.Key );
		}
		_objects.Clear ();
		while ( _pool.Count > 0 )
		{
			GameObject go = _pool.Dequeue ();
			if ( go != null )
				Destroy ( go );
		}
	}
}
```

**Comment by jordy-isaac** [comment]
Thanks for letting us know and for your feedback on this @demi180

**Comment by atripathi-mb** [comment]
@brnkhy Could you take a look please?
@demi180 Thanks for reporting this issue.

**Comment by demi180** [comment]
@brnkhy yes, you're right. Our particular case isn't moving around the map, we're just using one chunk at a time. I'd be just as happy to keep all the cached data on reload.

I could probably just add a DontDestroy and a check to our map object to circumvent this problem entirely, now that I think about it. I can even do that without modifying any of your scripts. I have no specific need to re-randomize the building textures and tree placement. In fact it'd be easier for the trees to stay the same in this case, since we've got Python scripts that have to receive all the collision information...
Anyway thanks again!

**Comment by brnkhy** [comment]
@demi180 glad to hear it works well for now. We'll probably look into this soon as a part of bigger review/refactor so won't rush a fix if everything works for you now.
